src/wallet/output_alias.c

```diff
@@ -0,0 +1,423 @@
+  if (!state) {
+    printf("[%s:%d] unable to create state controller address unlock condition\n", __func__, __LINE__);
+    return -1;
+  }
+
+  unlock_cond_blk_t* governor = cond_blk_governor_new(govern_addr);
+  if (!governor) {
+    cond_blk_free(state);
+    printf("[%s:%d] unable to create governor address unlock condition\n", __func__, __LINE__);
+    return -1;
+  }
+
+  cond_blk_list_t* cond_blocks = cond_blk_list_new();
+  if (cond_blk_list_add(&cond_blocks, state) != 0) {
+    cond_blk_free(state);
+    cond_blk_free(governor);
+    cond_blk_list_free(cond_blocks);
+    printf("[%s:%d] add unlock condition failed\n", __func__, __LINE__);
+    return -1;
+  }
+  if (cond_blk_list_add(&cond_blocks, governor) != 0) {
+    cond_blk_free(state);
+    cond_blk_free(governor);
+    cond_blk_list_free(cond_blocks);
+    printf("[%s:%d] add unlock condition failed\n", __func__, __LINE__);
+    return -1;
+  }
+
+  output_alias_t* alias_output =
+      output_alias_new(amount, NULL, alias_id, state_index, NULL, 0, 0, cond_blocks, NULL, NULL);
+  if (!alias_output) {
+    cond_blk_free(state);
+    cond_blk_free(governor);
+    cond_blk_list_free(cond_blocks);
+    printf("[%s:%d] creating alias output failed\n", __func__, __LINE__);
+    return -1;
+  }
+
+  if (tx_essence_add_output(essence, OUTPUT_ALIAS, alias_output) != 0) {
+    cond_blk_free(state);
+    cond_blk_free(governor);
+    cond_blk_list_free(cond_blocks);
+    output_alias_free(alias_output);
+    return -1;
+  }
+
+  cond_blk_free(state);
+  cond_blk_free(governor);
+  cond_blk_list_free(cond_blocks);
+  output_alias_free(alias_output);
+
+  return 0;
+}
+
+int wallet_alias_create_send(iota_wallet_t* w, address_t* sender_addr, ed25519_keypair_t* sender_keypair,
+                             uint64_t const send_amount, address_t* state_ctrl_addr, address_t* govern_addr,
+                             address_t* alias_addr, res_send_message_t* msg_res) {
+  if (w == NULL || sender_addr == NULL || sender_keypair == NULL || state_ctrl_addr == NULL || govern_addr == NULL ||
+      alias_addr == NULL || msg_res == NULL) {
+    printf("[%s:%d] invalid parameters\n", __func__, __LINE__);
+    return -1;
+  }
+
+  int ret = 0;
+  utxo_outputs_list_t* unspent_outputs = NULL;
+  signing_data_list_t* sign_data = signing_new();
+
+  // create a tx
+  transaction_payload_t* tx = tx_payload_new(w->network_id);
+  if (!tx) {
+    printf("[%s:%d] create tx payload failed\n", __func__, __LINE__);
+    ret = -1;
+    goto end;
+  }
+
+  // get outputs from the sender address
+  uint64_t total_unspent_amount = 0;
+  unspent_outputs = wallet_get_unspent_basic_outputs(w, sender_addr, sender_keypair, send_amount, tx->essence,
+                                                     &sign_data, &total_unspent_amount);
+  if (!unspent_outputs) {
+    printf("[%s:%d] address does not have any unspent basic outputs\n", __func__, __LINE__);
+    ret = -1;
+    goto end;
```

Comment:
tx is not freed at the end.
```suggestion
   tx_payload_free(tx);
    ret = -1;
    goto end;
```